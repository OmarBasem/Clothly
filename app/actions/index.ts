import {fetchUserAction} from "./user";
import {fetchProductsAction} from "./products";

export {fetchUserAction, fetchProductsAction}