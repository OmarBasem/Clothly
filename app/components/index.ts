import Title from "./Title";
import Back from './Back';
import TopBar from "./TopBar";
import SearchBar from "./SearchBar";
import Product from "./Product";
import ProductsRow from "./ProductsRow";


export {
    Title,
    Back,
    SearchBar,
    TopBar,
    Product,
    ProductsRow
}