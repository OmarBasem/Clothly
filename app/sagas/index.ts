import {all} from 'redux-saga/effects';

import user from './user'
import products from "./products";

const root = function* root() {
    yield all([
        user(),
        products()
    ]);
};

export default root;