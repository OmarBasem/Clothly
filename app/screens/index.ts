import HomeScreen from "./Home/HomeScreen";
import ProductScreen from "./Home/ProductScreen";
import CartScreen from "./Cart/CartScreen";
import ProfileScreen from "./Profile/ProfileScreen";
import WishlistScreen from "./Wishlist/WishlistScreen";

export {
    HomeScreen,
    ProductScreen,
    CartScreen,
    ProfileScreen,
    WishlistScreen
}