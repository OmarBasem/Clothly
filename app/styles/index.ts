import {DefaultTheme} from "@react-navigation/native";

export const theme = {
    mainColor: '#1C284F'
}


export const navTheme = {
    ...DefaultTheme,
    colors: {
        ...DefaultTheme.colors,
        background: '#fff'
    },
};
