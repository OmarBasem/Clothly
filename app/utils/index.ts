import {Dimensions} from "react-native";

export const sliceArrIntoRows = (data: Array<any>, n: number) => {
    const result = [];
    let temp = [];
    for (let i = 0; i < data.length; ++i) {
        if (i > 0 && i % n === 0) {
            result.push(temp);
            temp = [];
        }
        temp.push(data[i]);
    }
    if (temp.length > 0) {
        while (temp.length !== n) {
            temp.push(null);
        }
        result.push(temp);
    }
    return result;
};

export const W = (percentage: number) => Dimensions.get('window').width / 100 * percentage;
export const H = (percentage: number) => Dimensions.get('window').height / 100 * percentage;