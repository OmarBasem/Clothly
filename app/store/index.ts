import {configureStore} from '@reduxjs/toolkit'
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import rootReducer from '../reducers'
import AsyncStorage from '@react-native-async-storage/async-storage';
import createSagaMiddleware from 'redux-saga'
import sagas from "../sagas";

const persistConfig = {
    key: 'root',
    storage: AsyncStorage,
    whitelist: [] // persist the user data in the redux store
}

const persistedReducer = persistReducer(persistConfig, rootReducer)
const sagaMiddleware = createSagaMiddleware()

export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
            },
        }).concat(sagaMiddleware),
})
sagaMiddleware.run(sagas)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const persistor = persistStore(store)


