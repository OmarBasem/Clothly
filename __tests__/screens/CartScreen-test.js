import React from 'react';
import renderer from "react-test-renderer";
import {CartScreen} from '../../app/screens'



describe('Testing Screens rendering', () => {
    test('CartScreen renders correctly', () => {
        const tree = renderer.create(<CartScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})