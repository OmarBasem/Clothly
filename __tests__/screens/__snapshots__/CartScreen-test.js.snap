// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Screens rendering CartScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="CartScreen"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "grey",
          "fontSize": 80,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‡¡
  </Text>
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Your cart is empty
  </Text>
</View>
`;
